### PR TITLE
Pattern: Usually use REM for spacing

### PR DESCRIPTION
**Pull request by SpiderMasterMind**

### CODE REVIEW

**Review by sebsogamoso — COMMENTED** [review]
I don't have any strong opinions about this one and I am okay with merging it. Nonetheless, unless we have some standard dimensions using REM that we can reuse across the site I think this may be a very hard convention to fully adopt.

Are there any additional resources, apart from the link at the bottom of this that would help us get acquainted with how to properly use REM instead of pixels?

File: _patterns/use-rem-for-spacing.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+Pixel units are fixed and do not scale with browser settings.
+
+EM units may sometimes be suitable but they are harder to reason about as their final calculation may be relative to complicated nesting logic. Some developers prefer to use EM for media queries but this becomes tricky when a max and min width is declared.
+
+## Exceptions
+
+It is usually necessary to use fixed Px dimensions for images/image containers so the images inside scale correctly.
+It is necessary to use fixed Px dimensions for 'line-height' so copy horizontal spacing scales correctly.
+
+> Take a second to see if the REM sizing you are using exists already as a variable, and if it doesn't it probably should, to maintain consistent styling on the site. Take a look in app/assets/stylesheets/bp2017/. Particularly: _spacing.scss, _typography.scss, _variables.scss _breakpoints.scss and _colors.scss
```
Comment (bockets) [inline _patterns/use-rem-for-spacing.md:17]: This note seems specific to our app. Were we trying to keep these patterns generic?

File: _patterns/use-rem-for-spacing.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+````scss
+.blog-article {
+  padding: 32px;
+}
+````
+
+### Bad
+````scss
+.author-name {
+ . font-size: 14px;
```
Comment (bockets) [inline _patterns/use-rem-for-spacing.md:36]:
```suggestion
  .font-size: 14px;
```

**Comment by SpiderMasterMind** [comment]
@sebsogamoso We have scss variable files on the site with REM units, we have tried to employ these whenever possible, the most common are:
app/assets/stylesheets/bp2017/_typography.scss
app/assets/stylesheets/bp2017/_spacing.scss
I have tried to use these where necessary and most of our stylesheets use the shared values in their directory.

There are many articles explaining rem vs px vs em online, the link I provided was the best one I could find (in the first page of Google searching)

**Comment by sebsogamoso** [comment]
> we have tried to employ these whenever possible, the most common are:
app/assets/stylesheets/bp2017/_typography.scss
app/assets/stylesheets/bp2017/_spacing.scss

@SpiderMasterMind thanks for pointing those out. I have a question that is out of the scope of this PR but falls under the umbrella of conventions. Is there any reason why we seem to constantly use names like `bp2017`, `refactored`, `new_gen`, and all sorts of time-constrained terminology to name our files and classes? Maybe we should open a PR for that as well.

**Comment by SpiderMasterMind** [comment]
@sebsogamoso This is a HUGE pet peeve of mine, and a great suggestion for a patterns repo entry. I'll put it on my TODO list :)

**Review by baerbradford — APPROVED** [review]

**Pull request merged**